packages/grid/_modules_/grid/components/tools/Preferences.tsx

```diff
@@ -0,0 +1,176 @@
+    >
+      {children}
+    </div>
+  );
+}
+const useStyles = makeStyles((theme: Theme) => ({
+  root: {
+    backgroundColor: theme.palette.background.paper,
+    width: 600,
+    minHeight: 200,
+    display: 'flex',
+    flexDirection: 'column',
+    '& .tab': {
+      minWidth: 50,
+    },
+    '& .chip': {
+      margin: 2,
+    },
+  },
+}));
+//TODO refactor tab to navigation with a showNav prop on the component
+//TODO Extract Panel component?
+export const PreferencesPanel = () => {
+  const classes = useStyles();
+  const apiRef = React.useContext(ApiContext);
+
+  const [gridState, setGridState, forceUpdate] = useGridState(apiRef!);
+  const [target, setTarget] = React.useState<Element | null>(null);
+
+  const icons = useIcons();
+  const filterIconElement = React.createElement(icons.columnFiltering!, {});
+
+  const hideTimeout = React.useRef<any>();
+  const hidePreferences = React.useCallback(() => {
+    setGridState((state) => ({ ...state, preferencePanel: { open: false } }));
+    forceUpdate();
+  }, [forceUpdate, setGridState]);
+
+  const hidePreferencesDelayed = React.useCallback(() => {
+    hideTimeout.current = setTimeout(() => hidePreferences(), 100);
+  }, [hidePreferences]);
+  // This is to prevent the preferences from closing when you open a select box, issue with MUI core V4 => Fixed in V5
+  const dontHidePanel = React.useCallback(() => {
+    setImmediate(() => clearTimeout(hideTimeout.current));
+  }, []);
+
+  useApiEventHandler(apiRef!, PREVENT_HIDE_PREFERENCES, dontHidePanel);
+
+  const changeTab = React.useCallback(
+    (event: React.ChangeEvent<{}>, newValue: PreferencePanelsValue) => {
+      setGridState((state) => ({
+        ...state,
+        preferencePanel: { ...state.preferencePanel, open: true, openedPanelValue: newValue },
+      }));
+      forceUpdate();
+    },
+    [forceUpdate, setGridState],
+  );
+
+  const updateColumnTarget = React.useCallback(
+    ({ open, targetField }: PreferencePanelState) => {
+      if (targetField && open) {
+        const headerCellEl = findHeaderElementFromField(
+          apiRef!.current!.rootElementRef!.current!,
+          targetField!,
+        );
+        setTarget(headerCellEl);
+      }
+    },
+    [apiRef],
+  );
+
+  React.useEffect(() => {
+    updateColumnTarget(gridState.preferencePanel);
+  }, [gridState.preferencePanel, updateColumnTarget]);
+
+  const isColumnsTabOpen =
+    gridState.preferencePanel.openedPanelValue === PreferencePanelsValue.columns;
+  const isFiltersTabOpen = !isColumnsTabOpen;
+
+  return (
+    <Popper
+      placement="bottom"
+      open={gridState.preferencePanel.open}
+      anchorEl={target || apiRef?.current.rootElementRef!.current}
+      style={{ position: 'relative' }}
+    >
+      <div style={{ position: 'absolute', left: 0, top: 0, zIndex: 10000 }}>
+        <IconButton color="primary" aria-label="Close" component="span" onClick={hidePreferences}>
+          <CloseIcon />
+        </IconButton>
+      </div>
```

Comment:
This is my number one UX issue so far playing with the filter. It took me some time to figure out how to hide this *** popup 😆 . And even now that I know how to do it, oh god I hate it 😇.
I think that we need 1. the `ClickAwayListener`, 2. handle the <kbd>Escape</kbd> key. I would propose we use a native select, for now (`<Select native />`).

Reply:
Can we have a proper fix instead of using the native select?

Reply:
I'm aware of two solutions:

1. native slider: https://material-ui.com/components/selects/#native-select, the UI is still OK
2. #441

I can't think of anything else.

Reply:
using Native for now